cpp/src/centrality/betweenness_centrality.cu

```diff
@@ -176,158 +284,192 @@ void BC<VT, ET, WT, result_t>::compute_single_source(VT source_vertex)
-    for (VT source_idx = 0; source_idx < number_of_sources; ++source_idx) {
-      VT source_vertex = sources[source_idx];
-      compute_single_source(source_vertex);
-    }
-  } else {  // Otherwise process every vertices
-    // NOTE: Maybe we could still use number of sources and set it to number_of_vertices?
-    //       It woudl imply having a host vector of size |V|
-    //       But no need for the if/ else statement
-    for (VT source_vertex = 0; source_vertex < number_of_vertices; ++source_vertex) {
-      compute_single_source(source_vertex);
-    }
+}
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::accumulate_edges(VT max_depth,
+                                                dim3 grid_configuration,
+                                                dim3 block_configuration)
+{
+  for (VT depth = max_depth; depth >= 0; --depth) {
+    edges_accumulation_kernel<VT, ET, WT, result_t>
+      <<<grid_configuration, block_configuration, 0, stream>>>(betweenness,
+                                                               number_of_vertices,
+                                                               graph.indices,
+                                                               graph.offsets,
+                                                               distances,
+                                                               sp_counters,
+                                                               deltas,
+                                                               depth);
   }
-  rescale();
+}
+
+template <typename VT, typename ET, typename WT, typename result_t>
+void BC<VT, ET, WT, result_t>::accumulate_vertices_with_endpoints(VT source_vertex,
+                                                                  VT max_depth,
+                                                                  dim3 grid_configuration,
+                                                                  dim3 block_configuration)
+{
+  for (VT depth = max_depth; depth > 0; --depth) {
+    endpoints_accumulation_kernel<VT, ET, WT, result_t>
+      <<<grid_configuration, block_configuration, 0, stream>>>(betweenness,
+                                                               number_of_vertices,
+                                                               graph.indices,
+                                                               graph.offsets,
+                                                               distances,
+                                                               sp_counters,
+                                                               deltas,
+                                                               depth);
+  }
+  add_reached_endpoints_to_source_betweenness(source_vertex);
+  add_vertices_dependencies_to_betweenness();
+}
+
+// Distances should contain -1 for unreached nodes,
+// FIXME: It seems to be quite a lot to be able to increase the score
```

Comment:
?

Reply:
Reformulated, it seemed quite lengthy for only adding a value at a single position in the output.